resource_detectors/lib/opentelemetry/resource/detectors/container.rb

```diff
@@ -0,0 +1,59 @@
+# frozen_string_literal: true
+
+# Copyright The OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+module OpenTelemetry
+  module Resource
+    module Detectors
+      # Container contains detect class method for determining container resource attributes
+      module Container
+        extend self
+
+        UUID_PATTERN = '[0-9a-f]{8}[-_]?[0-9a-f]{4}[-_]?[0-9a-f]{4}[-_]?[0-9a-f]{4}[-_]?[0-9a-f]{12}'
+        CONTAINER_PATTERN = '[0-9a-f]{64}'
+        CONTAINER_REGEX = /(?<container>#{UUID_PATTERN}|#{CONTAINER_PATTERN})(?:.scope)?$/
+        CGROUP_V1_PATH = '/proc/self/cgroup'
+        CGROUP_V2_PATH = '/proc/self/mountinfo'
+
+        # Detects container attributes and creates a {OpenTelemetry::SDK::Resources::Resource} with the specified attributes.
+        # If no container attributes could be determined an empty resource is returned
+        #
+        # @return [OpenTelemetry::SDK::Resources::Resource]
+        def detect
```

Comment:
I am not entirely certain what the spec says about error handling in Resource detectors but I recommend you consider adding error handling with diagnostic information that will clue folks into why this resource detector may be failing and avoid raising errors that would cause the application to crash at boot time.

Reply:
Will do so once I split this of into it's own separate gem (see comment above). However, I do believe that currently this resource detector will not crash if it cannot determine the `container.id`?